### PR TITLE
CLOUDP-80279: Add a confirmation message to Deauthorize a Cloud Provider Access Role

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-80279

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
 ./bin/mongocli atlas cloudProviders accessRoles aws deauthorize 5ffd8cc17bd5c049ef1c88f5
? Are you sure you want to deauthorize 5ffd8cc17bd5c049ef1c88f5 Yes
AWS IAM role successfully deauthorized.
```

```
 ./bin/mongocli atlas cloudProviders accessRoles aws deauthorize 5ffd898d15a4f913021b9305
? Are you sure you want to deauthorize 5ffd898d15a4f913021b9305 No
AWS IAM role not deauthorized.
```

### CODE REVIEW

File: internal/prompt/prompt.go [hunk 1]
```diff
@@ -28,6 +28,14 @@ func NewDeleteConfirm(entry string) survey.Prompt {
 	return prompt
 }
 
+// NewCustomizeConfirm creates a prompt to confirm if the entry should be deleted
+func NewCustomizeConfirm(message, entry string) survey.Prompt {
+	prompt := &survey.Confirm{
+		Message: fmt.Sprintf("%s %s", message, entry),
+	}
+	return prompt
+}
```
Comment (gssbzn) [inline internal/prompt/prompt.go]: [nit] let's make this `NewConfirm(message string)` and you can do the `fmt.Sprintf("%s %s", message, entry)` before calling the function

File: internal/cli/delete_opts.go [hunk 2]
```diff
@@ -83,6 +83,16 @@ func (opts *DeleteOpts) Prompt() error {
 	return survey.AskOne(p, &opts.Confirm)
 }
 
+// CustomizedPrompt confirms that the resource should be deleted
+func (opts *DeleteOpts) CustomizedPrompt(message string) error {
```
Comment (gssbzn) [inline internal/cli/delete_opts.go:87]: [nit] a better go name would be something like `PromptWithMessage(`

**Review by gssbzn — COMMENTED** [review]
left a couple of nits

File: internal/prompt/prompt.go [hunk 3]
```diff
@@ -28,6 +28,14 @@ func NewDeleteConfirm(entry string) survey.Prompt {
 	return prompt
 }
 
+// NewConfirm creates a prompt to confirm if the entry should be deleted
+func NewConfirm(message, entry string) survey.Prompt {
+	prompt := &survey.Confirm{
+		Message: fmt.Sprintf(message, entry),
+	}
+	return prompt
+}
```
Comment (gssbzn) [inline internal/prompt/prompt.go:37]: sorry if this was not clear but with the rename is to also just take the message
```suggestion
func NewConfirm(message string) survey.Prompt {
	prompt := &survey.Confirm{
		Message: message,
	}
	return prompt
}
```
and then when you call `NewConfirm` you do it like `NewConfirm(fmt.Sprintf(message, entry))`
You don't need to couple confirm and your custom format, also you could directly do 

```
&survey.Confirm{
		Message: fmt.Sprintf(message, entry),
	}
```
inside `PromptWithMessage`

**Comment by andreaangiolillo** [comment]
@gssbzn ready for review 🔍

**Review by gssbzn — APPROVED** [review]
LGTM :shipit:

**Pull request merged**